File: src/main/java/me/alegian/thavma/impl/client/renderer/blockentity/PillarBER.java
package me.alegian.thavma.impl.client.renderer.blockentity;

import me.alegian.thavma.impl.client.renderer.geo.layer.EmissiveGeoLayer;
import me.alegian.thavma.impl.common.block.entity.PillarBE;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.api.distmarker.OnlyIn;
import software.bernie.geckolib.model.DefaultedBlockGeoModel;
import software.bernie.geckolib.renderer.GeoBlockRenderer;

import static me.alegian.thavma.impl.ThavmaKt.rl;

@OnlyIn(Dist.CLIENT)
public class PillarBER extends GeoBlockRenderer<PillarBE> {
  public PillarBER() {
    super(new DefaultedBlockGeoModel<>(rl("infusion_pillar")));
    this.addRenderLayer(new EmissiveGeoLayer<>(this));
  }
}

File: src/main/java/me/alegian/thavma/impl/client/renderer/blockentity/PillarBER.kt
package me.alegian.thavma.impl.client.renderer.blockentity

import com.mojang.blaze3d.vertex.PoseStack
import com.mojang.blaze3d.vertex.VertexConsumer
import me.alegian.thavma.impl.client.renderer.geo.layer.EmissiveGeoLayer
import me.alegian.thavma.impl.common.block.entity.PillarBE
import me.alegian.thavma.impl.init.registries.T7BlockStateProperties.MASTER
import me.alegian.thavma.impl.rl
import net.minecraft.client.renderer.MultiBufferSource
import net.minecraft.client.renderer.RenderType
import software.bernie.geckolib.cache.`object`.BakedGeoModel
import software.bernie.geckolib.model.DefaultedBlockGeoModel
import software.bernie.geckolib.renderer.GeoBlockRenderer

class PillarBER : GeoBlockRenderer<PillarBE>(DefaultedBlockGeoModel(rl("infusion_pillar"))) {
  init {
    addRenderLayer(EmissiveGeoLayer(this))
  }

  override fun actuallyRender(poseStack: PoseStack, be: PillarBE, model: BakedGeoModel, renderType: RenderType?, bufferSource: MultiBufferSource, buffer: VertexConsumer?, isReRender: Boolean, partialTick: Float, packedLight: Int, packedOverlay: Int, colour: Int) {
    if(!be.blockState.getValue(MASTER)) return
    super.actuallyRender(poseStack, be, model, renderType, bufferSource, buffer, isReRender, partialTick, packedLight, packedOverlay, colour)
  }

  override fun getRenderBoundingBox(be: PillarBE) = be.renderAABB
}
